### PR TITLE
 ci(language-service): enable language service tests in bazel

### PR DESCRIPTION
**Pull request by chuckjaz**
## PR Type
What kind of change does this PR introduce?

```
[x] CI related changes
```

## What is the current behavior?

Tests for `@angular/language-service` only run in `test.sh`.

## What is the new behavior?

Tests  for `@angular/language-service` in `bazel test`.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/compiler-cli/src/diagnostics/typescript_symbols.ts [hunk 1]
```diff
@@ -512,6 +512,9 @@ class PipesTable implements SymbolTable {
   values(): Symbol[] { return this.pipes.map(pipe => new PipeSymbol(pipe, this.context)); }
 }
 
+// This matches .d.ts files that look like ".../<package-name>/<package-name>.d.ts",
+const INDEX_PATTERN = /[\\|/](\w+)[\\|/]\1\.d\.ts$/;
```
Comment (vicb) [inline packages/compiler-cli/src/diagnostics/typescript_symbols.ts]: funny thing: you do not match `package-name` :)

File: packages/language-service/test/test_utils.ts [hunk 2]
```diff
@@ -75,12 +76,19 @@ export class MockTypescriptHost implements ts.LanguageServiceHost {
...
-    let angularIndex = moduleFilename.indexOf('@angular');
-    if (angularIndex >= 0)
-      this.angularPath = moduleFilename.substr(0, angularIndex).replace('/all/', '/all/@angular/');
-    let distIndex = moduleFilename.indexOf('/dist/all');
-    if (distIndex >= 0)
-      this.nodeModulesPath = myPath.join(moduleFilename.substr(0, distIndex), 'node_modules');
+    if (isInBazel()) {
+      const support = setup();
+      this.angularPath = path.join(support.basePath, 'node_modules/@angular');
+      this.nodeModulesPath = path.join(support.basePath, 'node_modules');
```
Comment (vicb) [inline packages/language-service/test/test_utils.ts]: compute the second one first and `.join('@angular')` ?

File: packages/language-service/test/test_utils.ts [hunk 3]
```diff
@@ -75,12 +76,19 @@ export class MockTypescriptHost implements ts.LanguageServiceHost {
...
+    } else {
+      let angularIndex = moduleFilename.indexOf('@angular');
```
Comment (vicb) [inline packages/language-service/test/test_utils.ts]: const

File: packages/language-service/test/test_utils.ts [hunk 4]
```diff
@@ -75,12 +76,19 @@ export class MockTypescriptHost implements ts.LanguageServiceHost {
...
+      if (angularIndex >= 0)
+        this.angularPath =
+            moduleFilename.substr(0, angularIndex).replace('/all/', '/all/@angular/');
+      let distIndex = moduleFilename.indexOf('/dist/all');
```
Comment (vicb) [inline packages/language-service/test/test_utils.ts]: const

File: packages/compiler-cli/src/diagnostics/typescript_symbols.ts [hunk 5]
```diff
@@ -600,7 +603,18 @@ class PipeSymbol implements Symbol {
...
-  const sourceFile = context.program.getSourceFile(type.filePath);
+  let sourceFile = context.program.getSourceFile(type.filePath);
+  if (!sourceFile) {
+    // This handles a case where an <packageName>/index.d.ts and a <packageName>/<packageName>.d.ts
+    // are in the same directory. If we are looking for <packageName>/<packageName> and didn't
+    // find it, look for <packageName>/index.d.ts as the program might have found that instead.
+    const p = type.filePath as string;
+    const m = p.match(INDEX_PATTERN);
+    if (m) {
+      const indexVersion = path.join(p.substr(0, m.index ! + m[1].length + 1), 'index.d.ts');
```
Comment (vicb) [inline packages/compiler-cli/src/diagnostics/typescript_symbols.ts]: I find `join(basename() + 'index.d.ts';` easier to understand

File: packages/compiler-cli/src/diagnostics/typescript_symbols.ts (hunk 1 above)
Comment (vicb) [reply packages/compiler-cli/src/diagnostics/typescript_symbols.ts]: `([^\|/]+)` ?

**Review by vicb — CHANGES_REQUESTED** [review]
`INDEX_PATTERN` should be improved

File: packages/compiler-cli/src/diagnostics/typescript_symbols.ts (hunk 1 above)
Comment (chuckjaz) [reply packages/compiler-cli/src/diagnostics/typescript_symbols.ts]: I am not sure what you are asking. I need to match `/foo/foo.d.ts` but not `/foo/bar.d.ts`.

**Review by vicb — APPROVED** [review]

**Pull request closed without merging**